tests/test_integration.py

```diff
@@ -530,6 +529,85 @@ def create_layout():
         self.startServer(app)
         time.sleep(0.5)
 
+    def test_multi_output(self):
+        app = dash.Dash(__name__)
+        app.scripts.config.serve_locally = True
+
+        app.layout = html.Div([
+            html.Button('OUTPUT', id='output-btn'),
+
+            html.Table([
+                html.Thead([
+                    html.Tr([
+                        html.Th('Output 1'),
+                        html.Th('Output 2')
+                    ])
+                ]),
+                html.Tbody([
+                    html.Tr([html.Td(id='output1'), html.Td(id='output2')]),
+                ])
+            ]),
+
+            html.Div(id='output3'),
+            html.Div(id='output4')
+        ])
+
+        @app.callback([Output('output1', 'children'), Output('output2', 'children')],
+                      [Input('output-btn', 'n_clicks')],
+                      [State('output-btn', 'n_clicks_timestamp')])
+        def on_click(n_clicks, n_clicks_timestamp):
+            if n_clicks is None:
+                raise PreventUpdate
+
+            return n_clicks, n_clicks_timestamp
+
+        # Dummy callback for DuplicateCallbackOutput test.
+        @app.callback(Output('output3', 'children'),
+                      [Input('output-btn', 'n_clicks')])
+        def dummy_callback(n_clicks):
+            if n_clicks is None:
+                raise PreventUpdate
+
+            return 'Output 3: {}'.format(n_clicks)
+
+        # Test that a multi output can't be included in a single output
+        with self.assertRaises(DuplicateCallbackOutput) as context:
+            @app.callback(Output('output1', 'children'),
+                          [Input('output-btn', 'n_clicks')])
+            def on_click_duplicate(n_clicks):
+                if n_clicks is None:
+                    raise PreventUpdate
+
+                return 'something else'
+
+        self.assertTrue('output1' in context.exception.args[0])
+
+        # Test a multi output cannot contain a used single output
```

Comment:
Two more cases I don't see covered (unless I'm missing it somewhere):
- you can't use the same output twice in one callback
- you can't use the same output in two overlapping multi-output sets (eg `#o1.value,#o2.value` and `#o1.value,#o3.value`